[Transform] Implement per-transform num_failure_retries setting. (#87361)
elastic/elasticsearch: docs/reference/settings/transform-settings.asciidoc

exhausted, the {transform} task is marked as `failed`. The default value is `10`
with a valid minimum of `0` and maximum of `100`. If a {transform} is already
running, it has to be restarted to use the changed setting.

> Please keep in mind that there is also a possibility to specify
> `num_failure_retries` setting on an individual transform level which has the
> same effect and should generally be preferred.

Reviewer: ```suggestion
The `num_failure_retries` setting can also be specified on an individual {transform} level. Specifying 
this setting for each {transform} individually is recommended.
```

Author: Done.
